### PR TITLE
Fetch missing records (only unintelligent batch of 500)

### PR DESCRIPTION
**Pull request by mgranberry**
I haven't done any intelligent selection of spans to update, and it should probably send an intent to the activity to trigger a redraw when the fetch finishes.

### CODE REVIEW

File: mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/Bg.java [hunk 1]
```diff
@@ -224,7 +248,7 @@ public int ageLevel() {
     public static Bg last() {
         return new Select()
                 .from(Bg.class)
-                .orderBy("_ID desc")
+                .orderBy("datetime desc")
```
Comment (StephenBlackWasAlreadyTaken) [inline mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/Bg.java]: goof call!

**Comment by StephenBlackWasAlreadyTaken** [comment]
Looks great!
The storing of calibrations and use of raw data isnt in here yet correct (or did I just miss it)? In any event Im ready to merge assuming Im not missing anything

**Comment by mgranberry** [comment]
The raw data changes accidentally made it in. 

It looks like there are some duplicated records that are making it in when there is both a pebble endpoint source and broadcasted xdrip records that I would like to get figured out before this gets put in front of too many people.

**Comment by StephenBlackWasAlreadyTaken** [comment]
Ah, I have a fix for that already I just never pushed it up, one sec

**Comment by StephenBlackWasAlreadyTaken** [comment]
yrg, I forgot its in the massive share support branch
you just need this line though https://github.com/StephenBlackWasAlreadyTaken/NightWatch/commit/3c9b3e7270da2f7a4eb107eded5c6e5c97d0d1d6#diff-e1cfba529ff86b9d2d762509ca8fde40R43

**Comment by mgranberry** [comment]
With the SharedPreferences bugfix, this works well for me.  Feel free to merge or let me know if you see anything.

**Comment by StephenBlackWasAlreadyTaken** [comment]
Merging it!

File: mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/dexdrip/IntentService.java [hunk 2]
```diff
@@ -34,13 +34,13 @@ protected void onHandleIntent(Intent intent) {
                 if (bgEstimate == 0)
                     return;
 
-                int battery = (int) (100 * intent.getIntExtra(Intents.EXTRA_SENSOR_BATTERY, 0) / 255f);
+                int battery = intent.getIntExtra(Intents.EXTRA_SENSOR_BATTERY, 0);
```
Comment (StephenBlackWasAlreadyTaken) [inline mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/dexdrip/IntentService.java]: https://github.com/StephenBlackWasAlreadyTaken/NightWatch/issues/15

**Comment by StephenBlackWasAlreadyTaken** [comment]
https://github.com/StephenBlackWasAlreadyTaken/NightWatch/issues/2

File: mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/Rest.java [hunk 3]
```diff
@@ -21,6 +22,7 @@
 public class Rest {
     private Context mContext;
     private String mUrl;
+    private static final String UNITS = "mgdl";
```
Comment (StephenBlackWasAlreadyTaken) [inline mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/Rest.java]: https://github.com/StephenBlackWasAlreadyTaken/NightWatch/issues/16

**Pull request merged**